core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -238,6 +240,8 @@ public String toString() {
   private final Map<Integer, SortOrder> sortOrdersById;
   private final List<HistoryEntry> snapshotLog;
   private final List<MetadataLogEntry> previousFiles;
+  private final Map<String, SnapshotRef> refs;
+  private final SetMultimap<Long, SnapshotRef> refsById;
```

Comment:
I don't understand why these are indexed by snapshot rather than name. Is that a common pattern?